lib/datadog/tracing.rb

```diff
@@ -115,31 +194,10 @@ def log_correlation
       # {.shutdown!} cannot be reversed.
       # @public_api
       def shutdown!
-        return unless tracer
+        current_tracer = tracer
+        return unless current_tracer
 
-        tracer.shutdown!
-      end
-
-      # The global integration registry.
-      #
-      # This registry holds a reference to all integrations available
-      # to the tracer.
-      #
-      # Integrations registered in the {.registry} can be activated as follows:
-      #
-      # ```
-      # Datadog::Tracing.configure do |c|
-      #   c.use :my_registered_integration, **my_options
-      # end
-      # ```
-      #
-      # New integrations can be registered by implementing the {Datadog::Contrib::Integration} interface.
-      #
-      # @return [Datadog::Contrib::Registry]
-      # @!attribute [r] registry
-      # @public_api
-      def registry
```

Comment:
What's up with removing the registry?

Reply:
It's not gone, it just exists in `contrib/extensions.rb`, as it always has.

The reason for this delineation is that `Contrib` will likely at some point become its own package, so users have the choice of whether they want to load all of these integrations (which they may not want to use.)

`registry` itself is a feature only used by integration authors anyways, so it's okay if it's not immediately visible in `Tracing`.

Reply:
Hmm... But shouldn't the registry still be considered a part of the core?

I'm thinking that even when we separate out all of the existing integrations, a customer that only wants to use the tracer + their own custom integration would need the registry.

Otherwise, they'd need to pull in the "contrib" just to be able to register their own integrations.

Reply:
While your point is fair, I don't see the benefit of having registry code in the core if it isn't used.

In the scenario where a user uses only the tracer core and a custom integration not a part of `Contrib`, you can do it one of a couple of ways. (1) The most likely is they don't use integration framework at all, they use manual instrumentation as they please. (2) You can make `Contrib` have it's own base (loaded by a `require`) which can be loaded independently of the integrations, such that you'd add the Contrib package, but only load the base within it.

I really don't think this case is common, but there are reasonable ways to handle this. I think it's more important to have a slim, minimal core, because it is more common to want the core sans integrations than this case.

Reply:
My concern is maybe clearer in the actual registry file -- it's because having it outside of the core increases complexity a lot because then it needs to monkey patch itself into the core.

As for slimness, does it make such a difference? It looks like we're discussing a couple of files, and a couple more classes and instances in memory, sounds like something in the kilobytes range (?).

Reply:
I just don't see them as belonging together; registry is a feature of integrations, and only exists for the purpose of integrations.

To give the benefit of the doubt, if the registry were used additionally by something other than tracing, then you could argue integrations are not "tracing integrations" but all-purpose Datadog integrations. Hypothetically, in this scenario, `Contrib` would be a first class package that uses `Datadog`, and would host a "Rails" integration that would include feature support for tracing, logging, and whatever else we can provide for that particular gem.

Although it would be more compelling to put something in the core that's used by two products, I still think there's a strong argument for saying `Contrib` are "community extensions" and should be treated as an externality.